datapipelines/src/main/kotlin/io/customer/datapipelines/DataPipelinesModule.kt

```diff
@@ -75,6 +77,84 @@ internal constructor(
     override fun initialize() {
     }
 
+    // Gets the userId registered by a previous identify call
+    // or null if no user is registered
+    private val registeredUserId: String? get() = analytics.userId()
+
+    /**
+     * Common method to identify a user profile with traits.
+     * The method is responsible for identifying the user profile with the given traits
+     * and running any necessary hooks.
+     * All other identify methods should call this method to ensure consistency.
+     * For methods that don't require to mention traits, use [Nothing] as the type.
+     */
+    private fun <Traits> commonIdentify(
+        userId: String,
+        traitsMap: CustomAttributes? = null,
+        traitsSerializable: Pair<Traits, SerializationStrategy<Traits>>? = null
+    ) {
+        if (userId.isBlank()) {
+            logger.debug("Profile cannot be identified: Identifier is blank. Please retry with a valid, non-empty identifier.")
+            return
+        }
+
+        logger.info("identify profile with id $userId")
+        when {
+            traitsMap != null -> {
+                logger.debug("identify profile with traits $traitsMap")
+                analytics.identify(
+                    userId = userId,
+                    traits = traitsMap
+                )
+            }
+
+            traitsSerializable != null -> {
+                logger.debug("identify profile with traits $traitsSerializable")
+                analytics.identify(
+                    userId = userId,
+                    traits = traitsSerializable.first,
+                    serializationStrategy = traitsSerializable.second
+                )
+            }
+
+            else -> {
+                analytics.identify(userId = userId)
+            }
+        }
+    }
+
+    override var profileAttributes: CustomAttributes
+        get() = analytics.traits() ?: emptyMap()
+        set(value) {
+            val userId = registeredUserId
+            if (userId != null) {
+                commonIdentify<Nothing>(userId = userId, traitsMap = value)
+            } else {
+                logger.debug("No user profile found, updating traits for anonymous user ${analytics.anonymousId()}")
+                analytics.identify(traits = value)
+            }
+        }
+
+    override fun identify(userId: String) {
+        commonIdentify<Nothing>(userId = userId)
```

Comment:
`commonIdentify<Nothing>` might not be very clean, but this does help having a common method to identify profiles for all cases. Since this is internal, I'm happy continue using it the same way.

Reply:
hmm i think there exists a world where we don't have to utilize `Nothing` in the first place.